### DISCUSSION TITLE
ODBC multiple_statements returns result of first

### DISCUSSION THREAD

**Discussion by ccoffey** [jeremyevans/sequel#2117] [General]
@jeremyevans I don't know if this is an appropriate place to ask this question. I don't know if this behaviour is because of the Snowflake driver or the Sequel GEM. My apologies if it is not related to the Sequel GEM.

According to this Snowflake doc: https://docs.snowflake.com/en/developer-guide/odbc/odbc-using you can execute multiple statements together by doing the following:

First you need to alter the session to set `MULTI_STATEMENT_COUNT = 0`
```
db.run("alter session set MULTI_STATEMENT_COUNT = 0;")
```

Then you can execute multiple statements in a single request:
```
db.fetch("select 1;select 2;").all
=> [{:"1"=>1}]
```

The really odd thing is that the result of the first query is being returned. I was expecting to get the result of the last query as the output.

**Comment by jeremyevans** [comment]
Sequel does not support multiple statements in the same query (as a general rule across adapters).  Some drivers may support it, but in that case, results are undefined.

**Reply by ccoffey** [reply]
Thank you for your reply. Unfortunately I am trying to replace an existing client, that uses multi-statement queries, with sequel. 

By looking at the query history view in Snowflake, I can see that when I do execute a multi-statement query via sequel it seems to do the right thing:

![Screenshot 2024-01-09 at 17 33 48](https://github.com/jeremyevans/sequel/assets/456400/5b926d86-4568-4309-a100-e15f0d9152a5)

I think this might work, as long as I don't care about the output of a multi-statement query 🤔 

Actually one thing I really need to verify is what the undefined behaviour is if one of the n queries fails!

Thank you again for your reply 🙇

**Comment by ccoffey** [comment]
I know you said that the results are undefined. But is it possible that Sequel could return the result for the first query before the second is even executed? Or is there a guarantee that at least all queries will be executed before a result is returned?

**Reply by jeremyevans** [reply]
Undefined means undefined, there are no guarantees in this case. If supported at all, behavior may vary by adapter and driver.

**Reply by ccoffey** [reply]
Thank you for your reply. I understand
